python/pyarrow/tests/test_parquet.py

```diff
@@ -69,10 +66,7 @@ def _check_roundtrip(table, expected=None, **params):
         expected = table
 
     result = _roundtrip_table(table, **params)
-    if not result.equals(expected):
-        print(expected)
-        print(result)
-        assert result.equals(expected)
+    assert result.equals(expected)
 
     result = _roundtrip_table(result, **params)
```

Comment:
Does this intentionally check the roundtrip twice?

Reply:
Looking at where it was added https://github.com/apache/arrow/commit/3b69c5acafaccb211f232b586a4e0b1172809809, I think you can remove the redundant check

Reply:
done

Reply:
My memory is fuzzy, but I _think_ the point was to test twice.

Reply:
@pitrou you're right, I misread the roundtrip check, which is different from the previous one. @kszucs can you restore those lines and rebase?

Reply:
Sure, will do